flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistry.java

```diff
@@ -198,6 +204,14 @@ public boolean isShutdown() {
 	 * Shuts down this registry and the associated {@link MetricReporter}.
 	 */
 	public void shutdown() {
+		Future<Boolean> stopFuture = null;
```

Comment:
Because that way we can exploit the asynchronous stop operation of the actors without blocking the stopping of the reporters and the executors.